web/packages/core/src/ruffle-player.tsx

```diff
@@ -1596,6 +1596,45 @@ export class RufflePlayer extends HTMLElement {
             return;
         }
 
+        // Calculate the top-left position where the context menu should be drawn.
+        // TODO: Allow the context menu to escape the document body while being mindful of scrollbars.
+        const getContextMenuOrigin = (playerRect: DOMRect, contextMenuRect: DOMRect) => {
+            const overflowX = Math.max(
+                0,
+                event.clientX +
+                    contextMenuRect.width -
+                    document.documentElement.clientWidth,
+            );
+            const overflowY = Math.max(
+                0,
+                event.clientY +
+                    contextMenuRect.height -
+                    document.documentElement.clientHeight,
+            );
+
+            const x = event.clientX - playerRect.x - overflowX;
+            const y = event.clientY - playerRect.y - overflowY;
+            
+            return [x, y];
+        }
+
+        // Don't redraw context menu when right-clicking inside of it.
+        // This emulates the behavior of traditional context menus, and prevents
+        // cascading right-clicks from causing the context menu to escape the player.
+        if (!this.contextMenuOverlay.classList.contains("hidden")) {
+            const playerRect = this.getBoundingClientRect();
+            const contextMenuRect = this.contextMenuElement.getBoundingClientRect();
+            const [x, y] = getContextMenuOrigin(playerRect, contextMenuRect);
+
+            const minX = contextMenuRect.x - playerRect.x;
+            const minY = contextMenuRect.y - playerRect.y;
+            const maxX = minX + contextMenuRect.width;
+            const maxY = minY + contextMenuRect.height;
+
+            if (x! >= minX && x! <= maxX && y! >= minY && y! <= maxY)
+                return;
+        }
```

Comment:
We probably shouldn't perform low-level calculations of x,y positions, but use the event model instead. In this case the following should work in the constructor:
```ts
this.contextMenuElement.addEventListener("contextmenu", (event) => {
    event.stopPropagation();
    event.preventDefault();
});
```